pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -131,14 +144,53 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 	return externalMetrics
 }
 
+func makeChunks(batch []string) (chunks [][]string) {
+	for i := 0; i < len(batch); i += chunkSize {
+		if i+chunkSize > len(batch) {
+			chunks = append(chunks, batch[i:])
+			break
+		}
+		chunks = append(chunks, batch[i:i+chunkSize])
+	}
+	return chunks
+}
+
 // validateExternalMetric queries Datadog to validate the availability and value of one or more external metrics
 func (p *Processor) validateExternalMetric(emList map[string]custommetrics.ExternalMetricValue) (processed map[string]Point, err error) {
-	var batch []string
+	batch := []string{}
 	for _, e := range emList {
 		q := getKey(e.MetricName, e.Labels)
 		batch = append(batch, q)
 	}
-	return p.queryDatadogExternal(batch)
+	chunks := makeChunks(batch)
+	log.Tracef("List of batches %v", chunks)
+
+	// we have a number of chunks with `chunkSize` metrics.
+	responses := make(chan queryResponse, len(batch))
+	processed = make(map[string]Point)
+
+	var waitResp sync.WaitGroup
+	waitResp.Add(len(chunks))
+	for _, c := range chunks {
+		go func(chunk []string) {
+			defer waitResp.Done()
+			resp, err := p.queryDatadogExternal(chunk)
```

Comment:
are we sure this eventually ends? (timeout)

Reply:
from what I understand the client use now a timeout: https://github.com/DataDog/datadog-agent/pull/4617/files#diff-05ddad3890afefbc763f110785c0f537R157

Reply:
ok looking at the client code. retryTimeout is the max query duration with multiple retry.
the initial retry period is 500ms with an exponential backoff policy.
https://github.com/zorkian/go-datadog-api/blob/557151e02d5f4f5a9298c2608f92a3ddd8b36462/vendor/github.com/cenkalti/backoff/exponential.go#L75

Reply:
all good